docs/reference/setup/install/docker.asciidoc

```diff
@@ -0,0 +1,290 @@
+
+ifeval::["{release-state}"=="unreleased"]
+
+However, version {version} of Elasticsearch has not yet been released, so no Docker image is currently available for this version.
+
+endif::[]
+
+ifeval::["{release-state}"!="unreleased"]
+
+The Docker image can be retrieved with the following command:
+
+["source","sh",subs="attributes"]
+--------------------------------------------
+docker pull {docker-image}
+--------------------------------------------
+
+endif::[]
+
+[[docker-cli-run]]
+==== Running Elasticsearch from the command line
+
+[[docker-cli-run-dev-mode]]
+===== Development mode
+
+Elasticsearch can be quickly started for development or testing use with the following command:
+
+ifeval::["{release-state}"=="unreleased"]
+
+Version {version} of the Elasticsearch Docker image has not yet been released.
+
+endif::[]
+
+ifeval::["{release-state}"!="unreleased"]
+
+["source","sh",subs="attributes"]
+--------------------------------------------
+docker run -p 9200:9200 -e "http.host=0.0.0.0" -e "transport.host=127.0.0.1" {docker-image}
+--------------------------------------------
+
+endif::[]
+
+[[docker-cli-run-prod-mode]]
+===== Production mode
+
+[[docker-prod-prerequisites]]
+[WARNING]
+=========================
+
+The `vm_max_map_count` kernel setting needs to be set to at least `262144` for production use.
+Depending on your platform:
+
+* Linux
++
+The `vm_map_max_count` setting should be set permanently in /etc/sysctl.conf:
++
+[source,sh]
+--------------------------------------------
+$ grep vm.max_map_count /etc/sysctl.conf
+vm.max_map_count=262144
+----------------------------------
++
+To apply the setting on a live system type: `sysctl -w vm.max_map_count=262144`
++
+* OSX with https://docs.docker.com/engine/installation/mac/#/docker-for-mac[Docker for Mac]
++
+The `vm_max_map_count` setting must be set within the xhyve virtual machine:
++
+["source","sh"]
+--------------------------------------------
+$ screen ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/tty
+--------------------------------------------
++
+Log in with 'root' and no password.
+Then configure the `sysctl` setting as you would for Linux:
++
+["source","sh"]
+--------------------------------------------
+sysctl -w vm.max_map_count=262144
+--------------------------------------------
++
+* OSX with https://docs.docker.com/engine/installation/mac/#docker-toolbox[Docker Toolbox]
++
+The `vm_max_map_count` setting must be set via docker-machine:
++
+["source","sh"]
+--------------------------------------------
+docker-machine ssh
+sudo sysctl -w vm.max_map_count=262144
+--------------------------------------------
+=========================
+
+The following example brings up a cluster comprising two Elasticsearch nodes.
+To bring up the cluster, copy the contents of <<docker-prod-cluster-composefile,docker-compose.yml>> to a file named `docker-compose.yml` and just type:
```

Comment:
I've reworded this so that it doesn't sound too pedantic:

> To bring up the cluster, use the \<\<docker-prod-cluster-composefile,`docker-compose.yml`\>\> and just type: